### PR TITLE
rgw: fix memory leak in copy_obj_to_remote_dest

### PR DESCRIPTION
**Pull request by weiqiaomiao**
Signed-off-by: weiqiaomiao wei.qiaomiao@zte.com.cn

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -6826,6 +6826,7 @@ int RGWRados::copy_obj_to_remote_dest(RGWObjState *astate,
 
   ret = read_op.iterate(0, astate->size - 1, out_stream_req->get_out_cb());
   if (ret < 0)
+    delete out_stream_req;
     return ret;
```
Comment (gaowanlong) [inline src/rgw/rgw_rados.cc]: You should at least missed the " { } " here.

**Comment by weiqiaomiao** [comment]
thanks @gaowanlong . I fix it

**Comment by weiqiaomiao** [comment]
retest this please

**Comment by weiqiaomiao** [comment]
@mattbenjamin  ping

**Comment by yuriw** [comment]
ready for merge http://pulpito.ceph.com/yuriw-2017-07-12_19:38:07-rgw-wip-yuri-testing2_2017_7_12_2-distro-basic-smithi/

**Comment by cbodley** [comment]
jenkins test this please

**Comment by yehudasa** [comment]
@weiqiaomiao @oritwas see https://github.com/ceph/ceph/pull/16324, a fix in a related code (noticed it when reviewing this one).

**Pull request merged**